src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -1089,14 +1107,16 @@ void VM_Version::get_processor_features() {
     _has_intel_jcc_erratum = IntelJccErratumMitigation;
   }
 
-  char buf[1024];
+  char buf[2048] = {'\0'};
   int res = jio_snprintf(
               buf, sizeof(buf),
               "(%u cores per cpu, %u threads per core) family %d model %d stepping %d microcode 0x%x",
               cores_per_cpu(), threads_per_core(),
               cpu_family(), _model, _stepping, os::cpu_microcode_revision());
   assert(res > 0, "not enough temporary space allocated");
-  insert_features_names(buf + res, sizeof(buf) - res, _features_names);
+  insert_features_names(_features, buf + res, sizeof(buf) - res, _features_names);
```

Comment:
x86 is the only platform which uses `insert_features_names`. Other platforms rely on macros. Maybe it's time to do the same on x86?